python-sdk/src/astro/databases/mysql.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pandas as pd
+import sqlalchemy
+from airflow.providers.mysql.hooks.mysql import MySqlHook
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from astro.databases.base import BaseDatabase
+from astro.options import LoadOptions
+from astro.settings import MYSQL_SCHEMA
+from astro.table import BaseTable, Metadata
+from astro.utils.compat.functools import cached_property
+
+DEFAULT_CONN_ID = MySqlHook.default_conn_name
+if TYPE_CHECKING:  # pragma: no cover
+    pass
+
+
+class MysqlDatabase(BaseDatabase):
+    DEFAULT_SCHEMA = MYSQL_SCHEMA
+
+    _create_schema_statement: str = (
+        "CREATE SCHEMA IF NOT EXISTS {} "
+        "DEFAULT CHARACTER SET = 'utf8' DEFAULT COLLATE = "
+        "'utf8_unicode_ci'"
+    )
+
+    def __init__(
+        self,
+        conn_id: str = DEFAULT_CONN_ID,
+        table: BaseTable | None = None,
+        load_options: LoadOptions | None = None,
+    ):
+        super().__init__(conn_id)
+        self.table = table
+        self.load_options = load_options
+
+    @cached_property
+    def hook(self) -> MySqlHook:
+        """Retrieve Airflow hook to interface with the mysql database."""
+        conn = MySqlHook(mysql_conn_id=self.conn_id).get_connection(self.conn_id)
+        kwargs = {}
+        if conn.schema is None:
+            if (
+                self.table
+                and self.table.metadata
+                and (self.table.metadata.database or self.table.metadata.schema)
+            ):
+                kwargs.update({"schema": self.table.metadata.database or self.table.metadata.schema})
+            else:
+                kwargs.update({"schema": self.DEFAULT_SCHEMA})
+        return MySqlHook(mysql_conn_id=self.conn_id, **kwargs)
+
+    def populate_table_metadata(self, table: BaseTable) -> BaseTable:
+        """
+        Given a table, check if the table has metadata.
+        If the metadata is missing, and the database has metadata, assign it to the table.
+        If the table schema was not defined by the end, retrieve the user-defined schema.
+        This method performs the changes in-place and also returns the table.
+        For mysql - schema is synonymous with database. So if the user supplies either
+        database or schema consider it to be the same
+
+        :param table: Table to potentially have their metadata changed
+        :return table: Return the modified table
+        """
+        if table.metadata and table.metadata.is_empty() and self.default_metadata:
```

Comment:
It seems odd that we set both a database and a schema in this function but then only check for one or the other in the hook function

Reply:
fixed this in latest commit